src/proto/uprotocol/uresource.rs

```diff
@@ -25,6 +30,41 @@ impl UResource {
     pub fn get_instance(&self) -> Option<&str> {
         self.instance.as_deref()
     }
+
+    /// Returns whether a `UResource` satisfies the requirements of a micro form URI
+    ///
+    /// # Returns
+    /// Returns a `Result<(), ValidationError>` where the ValidationError will contain the reasons it failed or OK(())
+    /// otherwise
+    ///
+    /// # Errors
+    ///
+    /// Returns a `ValidationError` in the failure case
+    pub fn validate_micro_form(&self) -> Result<(), ValidationError> {
+        let mut validation_errors = Vec::new();
+
+        if let Some(id) = self.id {
+            if id & URESOURCE_ID_VALID_BITMASK != 0 {
+                validation_errors.push(ValidationError::new(
+                    "ID does not fit within allotted 16 bits in micro form",
+                ));
+            }
+        } else {
+            validation_errors.push(ValidationError::new("ID must be present"));
+        }
+
+        if !validation_errors.is_empty() {
```

Comment:
see previous comment

Reply:
Yup, I agree, this could be made to bail earlier using `?`.